### PR TITLE
Gl/add cgi executable

### PR DESCRIPTION
**Pull request by gleal42**
### Small PR to add cgi executables.

I adjusted the CGIHandler class accordingly. 

Maybe we could even delete CGIHandler static variables and replace them with the config logic now.

What do you think?

Because CGI doesn't really exist in NGINX I'm not sure if the executable validation should be done at config level or later.

### CODE REVIEW

**Comment by mlrcbsousa** [comment]
> I adjusted the CGIHandler class accordingly.

Noice thanks

> Maybe we could even delete CGIHandler static variables and replace them with the config logic now.

what would this mean more specifically?

> Because CGI doesn't really exist in NGINX I'm not sure if the executable validation should be done at config level or later.

Also not sure I'm following, seems to me should be done at config level, what sort of validation are you talking about?

File: src/utils.cpp [hunk 1]
```diff
@@ -462,3 +462,13 @@ void			directory_indexing_resolve( URI & uri, const std::string &root, const Ser
 		throw HTTPStatus<404>();
 	uri.path = (*index);
 }
+
+void			URI::clear( void )
```
Comment (mlrcbsousa) [inline src/utils.cpp:466]: should be method in URI class in its own file

File: scripts/manual.sh [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-#    Updated: 2022/09/05 22:49:06 by gleal            ###   ########.fr        #
+#    Updated: 2022/09/08 23:21:33 by gleal            ###   ########.fr        #
 #                                                                              #
 # **************************************************************************** #
 
-docker build -f Dockerfile_gl -t fred_server .
-docker run --rm -it --name fred_server -v $(PWD):/usr/src
-/webserv -p 80:80 fred_server
+docker build -t fred_server .
+docker run --rm --security-opt seccomp=unconfined -it --name fred_server -v $(PWD):/usr/src/webserv -p 8080:8080 fred_server
```
Comment (mlrcbsousa) [inline scripts/manual.sh:14]: `fred_server` is a name that doesnt get across the purpose of the thing

File: Dockerfile [hunk 3]
```diff
@@ -14,5 +14,7 @@ COPY . /usr/src/webserv
 # Compile
 RUN make
 
+EXPOSE 8080
```
Comment (mlrcbsousa) [inline Dockerfile:17]: why is this needed? in order to have this variable dynamic with the e2e tests / matching the config, we would need to get it from the config files somehow

**Review by mlrcbsousa — CHANGES_REQUESTED** [review]
LGTM

Couple of things though. Please keep only actual test related files in the test folder. Unit tests, end-to-end tests, nginx compare tests, etc. Implementation files like the binaries should be in their own codebase domain, for the case of the binaries I would go with a `bin/` fodler

File: Dockerfile (hunk 3 above)
Comment (gleal42) [reply Dockerfile:17]: Pois, não me estava a reconhecer of requests então achei que fosse isto. Posso double check a ver se existe outra maneira.

**Pull request merged**